File: frontend/src/types/prediction.ts
export type ForecastRequest = {
  months_ahead?: number;
};

export type ForecastItem = {
  month: string;
  predicted_expense: number;
};

export type ExpenseForecastResponse = {
  model_name: string;
  training_points: number;
  last_month_actual: number | null;
  next_month_prediction: number;
  forecast: ForecastItem[];
  method: 'linear_regression' | 'moving_average' | 'no_history';
  confidence: 'low' | 'medium' | 'high';
  message: string;
};

File: backend/app/schemas/prediction.py
from datetime import date

from pydantic import BaseModel, ConfigDict, Field


class ForecastRequest(BaseModel):
    months_ahead: int = Field(default=3, ge=1, le=12)


class MonthlyForecastItem(BaseModel):
    month: date
    predicted_expense: float


class ExpenseForecastResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    model_name: str
    training_points: int
    last_month_actual: float | None
    next_month_prediction: float
    forecast: list[MonthlyForecastItem]
    method: str
    confidence: str
    message: str

